### ISSUE TITLE
[Bug report] vuepress broken with vite@2.9

### ISSUE THREAD

**Issue by zS1L3NT** [vuepress/vuepress-next#781]
## Workaround

Hi everyone, here is a possible workaround.

If you are using yarn, add  "resolution" in your package.json and force using `vite@2.8`, then everything should work fine.

```
{
 // ...
  "resolutions": {
    "vite": "2.8.6"
  },
}
```

If you are using npm, please make sure you are using `vite@2.8.x` by manualy declare `"vite": "2.8.6"` then a clean install. That should work in most cases.

We are actually not responsable for potential breaking changes from deps chain. 🙂 Give us some time to investigate and fix this.

-- Mr.Hope from VuePress Team

---

Here is the original issue:

### Description

VuePress was showing me this error on `useThemeLocaleData()`, so what I did was that I made a new branch and deleted all my project's dependencies and files, leaving only the VuePress code.

I deleted my `node_modules` folder and `pnpm-lock.yaml`, then did a `pnpm i` to install the `node_modules` and create the `pnpm-lock.yaml` again. However this still did not work and showed the same error.

### Steps to reproduce

You can clone the repository [here](https://github.com/zS1L3NT/ts-npm-validate-any/tree/vuepress-broken)

### Expected behavior

An empty VuePress app to run

### Screenshots

![Screenshot of the console](https://user-images.githubusercontent.com/26828488/160911557-adf28c49-72ab-4efb-8156-d66ed8620a62.png)

### Environment info

- Browser:

![Browser](https://user-images.githubusercontent.com/26828488/160912209-97933f02-e5d7-4c55-b333-24ba223525f5.png)

- Output of `vuepress info`:
```bash
  System:
    OS: Windows 10 10.0.22581
    CPU: (12) x64 AMD Ryzen 5 3600 6-Core Processor
    Memory: 2.95 GB / 15.95 GB
  Binaries:
    Node: 16.10.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.17 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.5.2 - C:\Program Files\nodejs\npm.CMD
  Utilities:
    Git: 2.34.1.
  Browsers:
    Chrome: Not Found
    Edge: Spartan (44.22581.100.0), Chromium (99.0.1150.55)
  npmPackages:
    @vuepress/bundler-vite:  2.0.0-beta.36
    @vuepress/bundler-webpack: Not Found
    @vuepress/cli:  2.0.0-beta.36
    @vuepress/client:  2.0.0-beta.35
    @vuepress/core:  2.0.0-beta.36
    @vuepress/markdown:  2.0.0-beta.36
    @vuepress/plugin-active-header-links:  2.0.0-beta.36
    @vuepress/plugin-back-to-top:  2.0.0-beta.36
    @vuepress/plugin-container:  2.0.0-beta.36
    @vuepress/plugin-docsearch: Not Found
    @vuepress/plugin-external-link-icon:  2.0.0-beta.36
    @vuepress/plugin-git:  2.0.0-beta.36
    @vuepress/plugin-google-analytics: Not Found
    @vuepress/plugin-medium-zoom:  2.0.0-beta.36
    @vuepress/plugin-nprogress:  2.0.0-beta.36
    @vuepress/plugin-palette:  2.0.0-beta.36
    @vuepress/plugin-prismjs:  2.0.0-beta.36
    @vuepress/plugin-pwa: Not Found
    @vuepress/plugin-pwa-popup: Not Found
    @vuepress/plugin-register-components: Not Found
    @vuepress/plugin-search: Not Found
    @vuepress/plugin-shiki: Not Found
    @vuepress/plugin-theme-data:  2.0.0-beta.36
    @vuepress/plugin-toc: Not Found
    @vuepress/shared:  2.0.0-beta.35
    @vuepress/theme-default:  2.0.0-beta.36
    @vuepress/utils:  2.0.0-beta.35
    vue:  3.2.31
    vue-loader: Not Found
    vue-router:  4.0.14
    vuepress: ^2.0.0-beta.36 => 2.0.0-beta.36
    vuepress-vite:  2.0.0-beta.36
    vuepress-webpack: Not Found
```

**Comment by dacodekid** [comment]
I can confirm the same issue started for me as well. 

<img width="744" alt="image" src="https://user-images.githubusercontent.com/10047844/160962701-1eee9af7-0e96-4620-9906-58b489adc9ae.png">

**Comment by Mister-Hope** [comment]
Vite released new version yesterday,may caused by that.

Deps tree may be broken, please double check whether you have different versions of @vue/xxxx and @vite/xxx packages installed.

**Comment by dacodekid** [comment]
@Mister-Hope ,  so far.. 

1. I deleted everything and started from scratch
2. Tried `pnpm` & `npm`
3. Tried to install `Beta 35 & 34` 

but none seems to work at this point. When the server starts, none of the packages (such as `vuepress-router`) gets started.

<img width="487" alt="image" src="https://user-images.githubusercontent.com/10047844/160966454-70c053e1-7fd8-48e3-8d2f-b599a2e8ee06.png">

**Comment by Mister-Hope** [comment]
> please double check whether you have different versions of @vue/xxxx and @vite/xxx packages installed, 

Please check this, also, you may try using vite@2.8 with `"vite": "~2.8.0"`.

**Comment by dacodekid** [comment]
@Mister-Hope I agree. I'm pretty sure it is `vite` that causing the trouble (a couple times, I saw a different error regards version numbers - but I didn't copy the error message and now I couldn't reproduce.  It was some like it was expecting version `12.x` but `14.x` was there) .  

Changing to `webpack` bundler works fine.  But I couldn't use `Vite 2.8.0` as even I install it manually, `vuepress` by default using `vite 2.9`.  Not sure how to force that.  I'll use `webpack` for time being.

```
devDependencies:
+ @vuepress/bundler-webpack 2.0.0-beta.36
+ vite 2.8.0 (2.9.0 is available)
+ vuepress 2.0.0-beta.36
+ webpack 5.70.0
```

```
vuepress dev docs

info Initializing VuePress and preparing data...

  vite v2.9.0 dev server running at:

  > Local:    http://localhost:8080/
  > Network:  http://192.168.1.138:8080/
```

**Comment by Mister-Hope** [comment]
> vuepress by default using vite 2.9

Nope

https://github.com/vuepress/vuepress-next/blob/main/packages/%40vuepress/bundler-vite/package.json#L43

You should be able to use latest vite@2.8.x

**Comment by Mister-Hope** [comment]
#784 should fix this

**Comment by Mister-Hope** [comment]
Hi everyone, here is a possible workaround.

If you are using yarn, add  "resolution" in your package.json and force using `vite@2.8`, then everything should be working fine.

```
{
 // ...
  "resolutions": {
    "vite": "2.8.2"
  },
}
```

We are actually not responsable for potential breaking changes from deps chain. 🙂 Give us some time to investigate and fix this.

**Comment by meteorlxy** [comment]
https://github.com/vitejs/vite/pull/7379 introduced some internal breaking changes, which broke one of the internal workaround plugin of vuepress-next.

I'll update the workaround plugin soon. It's still a workaround, and we need to find a way to solve it completely in the future.

**Comment by Mister-Hope** [comment]
New related bug #800. We will fix it soon.
